editor/editor_settings.cpp

```diff
@@ -475,9 +475,13 @@ void EditorSettings::_load_defaults(Ref<ConfigFile> p_extra_config) {
 
 	// External Programs
 	EDITOR_SETTING(Variant::STRING, PROPERTY_HINT_GLOBAL_DIR, "filesystem/external_programs/raster_image_editor", "", "")
+	add_property_hint(PropertyInfo(Variant::STRING, "filesystem/external_programs/raster_image_editor", PROPERTY_HINT_GLOBAL_FILE));
 	EDITOR_SETTING(Variant::STRING, PROPERTY_HINT_GLOBAL_DIR, "filesystem/external_programs/vector_image_editor", "", "")
+	add_property_hint(PropertyInfo(Variant::STRING, "filesystem/external_programs/vector_image_editor", PROPERTY_HINT_GLOBAL_FILE));
 	EDITOR_SETTING(Variant::STRING, PROPERTY_HINT_GLOBAL_DIR, "filesystem/external_programs/audio_editor", "", "")
+	add_property_hint(PropertyInfo(Variant::STRING, "filesystem/external_programs/audio_editor", PROPERTY_HINT_GLOBAL_FILE));
 	EDITOR_SETTING(Variant::STRING, PROPERTY_HINT_GLOBAL_DIR, "filesystem/external_programs/3d_model_editor", "", "")
+	add_property_hint(PropertyInfo(Variant::STRING, "filesystem/external_programs/3d_model_editor", PROPERTY_HINT_GLOBAL_FILE));
```

Comment:
Isn't the problem that you're using `PROPERTY_HINT_GLOBAL_DIR` when it should use `PROPERTY_HINT_GLOBAL_FILE` from  the start?
Can a property have two hints?

Reply:
🤦‍♂️
I completely missed that there is already a property hint here.